dataStructures/structures/priority_queue.js

```diff
@@ -0,0 +1,38 @@
+export default class PriorityQueue {
+  constructor() {
+    this.content = []
+    this.length = 0
+  }
+
+  enqueue(element, priority) {
+    if(priority < 1 || typeof priority !== 'number') {
+      throw new Error('This Priority isn\'t valid')
+    }
+    this.length += 1
+    this.content.push({priority, element})
+    this.content = this.content.sort((a, b) => a.priority > b.priority)
+  }
+
+  dequeue() {
+    if(this.length === 0) {return null}
```

Comment:
priority queue work like, things with a higher priority, get dequeued quicker? how does this check for priority?